SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -1138,7 +1138,7 @@ void USpatialActorChannel::RemoveRepNotifiesWithUnresolvedObjs(TArray<UProperty*
 			}
 
 			bool bIsSameRepNotify = RepLayout.Parents[ObjRef.Value.ParentIndex].Property == Property;
-			bool bIsArray = RepLayout.Parents[ObjRef.Value.ParentIndex].Property->ArrayDim > 1;
+			bool bIsArray = RepLayout.Parents[ObjRef.Value.ParentIndex].Property->ArrayDim > 1 || Cast<UArrayProperty>(Property) != nullptr;
```

Comment:
This allows arrays of structs to notify partial states. I would be in favor to make this behaviour optional.

Reply:
I agree, an option here would be good.

Reply:
After discussing with Tim, he made the point that some reference might never resolve because the local view has no interest in the object being referenced. So progressive notification seems to be the way to go. It may warrant the creation of some additional tool to either handle an array of refs as an event queue, or something that would guarantee that the receiver has interest over the references we are about to send it.